### PR TITLE
docs: document release process

### PR DESCRIPTION
**Pull request by matthieu-foucault**

### CODE REVIEW

File: docs/release.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+## 1 Release Pull Request
+
+- 1.1 Create a chore/release branch
+- 1.2 From the root of the repository, run `./app/node_modules/.bin/standard-version`. This will generate the [changelog](../CHANGELOG.md), increase the version number following [semantic versioning](https://semver.org/), commit the changes and tag that commit.
+- 1.3 note the version number printed by standard-version (referred to as &`<version>` below)
+- 1.4 run `cd schema/ && sqitch tag v<version>  -m "release v<version>"`
+- 1.5 In helm/cas-ciip-portal/Chart.yaml, set `appVersion` to `<version>`, and increas the `version` patch number.
+- 1.6 Commit with the chore(release) commit type.
+- 1.7 Create a pull request from chore/release to develop.
```
Comment (dleard) [inline docs/release.md:13]: I know standard-version does say to use the `--follow-tags` flag in the output, but it's something I missed the first time doing this, so maybe that should be mentioned here in the process docs?

**Review by dleard — DISMISSED** [review]

**Review by kriscooke — COMMENTED** [review]
@matthieu-foucault  Any chance you could add a link to this page in `docs/index.md`? That is the entrypoint to developer docs linked from the frontpage README.

**Comment by matthieu-foucault** [comment]
@kriscooke done. I also renamed `docs/index.md` to `docs/README.md` so that is shows up in the GitHub UI when you browse the `docs` folder

**Comment by kriscooke** [comment]
>  I also renamed docs/index.md to docs/README.md so that is shows up in the GitHub UI when you browse the docs folder

Ah, thank you! I knew I'd missed something basic like that

**Review by kriscooke — APPROVED** [review]

**Pull request merged**